fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -218,10 +226,27 @@ Table tbl = Table.builder(schemaName, tableName)
     .setTablespace(fhirTablespace)
     .addPrivileges(resourceTablePrivileges)
     .enableAccessControl(this.sessionVariable)
+    .addMigration(priorVersion -> {
+            List<IDatabaseStatement> statements = new ArrayList<>();
+                if (priorVersion == 1) {
+                    // Add statements here
+                }
+                return statements;
+            })
     .build(model);
 ```
 
-If a new resource is added to the specification, the schema utility automatically provisions it on the next execution of the update schema actions. 
+When the schema is applied for the first time, it creates the table (and related constructs) as according to the definition.
+When the schema is applied to an existing database, the framework will check the `FHIR_ADMIN.VERSION_HISTORY` table and, 
+if the version in the table is less than the version being applied, the framework will invoke the Migrations *instead*
+of calling CREATE with the table definition. Care should be taken to ensure that the migrated schema will match a freshly applied schema.
+See [Testing migrations](#testing-migrations) for information on verifying the fidelity of the schema migrations.
+
+NOTE: In Db2, certain alter table statements require a table REORG before the table becomes usable again. Additionally, the REORG
+may commit the current "unit of work" which can prevent the "all or nothing" semantics of the migration. For these reasons, its recommended
+to backup the database before invoking a migration and performing the migration off-line.
+
+In this way, if a new resource is added to the specification, the schema utility will automatically provision the corresponding table on the next execution of the fhir-persistence-schema `--update-schema` action. 
```

Comment:
```suggestion
When the schema is applied for the first time, it creates the table (and related constructs) as according to the definition.
When the schema is applied to an existing database, the framework checks the `FHIR_ADMIN.VERSION_HISTORY` table and, 
if the version in the table is less than the version being applied, the framework invokes the Migrations *instead*
of calling CREATE with the table definition. **Care** should be taken to ensure that the migrated schema matches a freshly applied schema.
See [Testing migrations](#testing-migrations) for information on verifying the fidelity of the schema migrations.

NOTE: In Db2, certain alter table statements require a table REORG before the table becomes usable again. Additionally, the REORG
may commit the current "unit of work" which can prevent the "all or nothing" semantics of the migration. For these reasons, its recommended
to backup the database before invoking a migration and performing the migration offline.

In this way, if a new resource is added to the specification, the schema utility will automatically provision the corresponding table on the next execution of the fhir-persistence-schema `--update-schema` action. 
```

Reply:
applied manually due to some other minor edits i had locally